core/devops.go

```diff
@@ -171,7 +171,14 @@ func (d *Devops) invokeOrQuery(ctx context.Context, chaincodeInvocationSpec *pb.
 	}
 
 	// Now create the Transactions message and send to Peer.
-	uuid := util.GenerateUUID()
+        var userGivenID = chaincodeInvocationSpec.UserGivenID
+        var uuid string
+        if userGivenID != "" {
+                uuid = userGivenID
```

Comment:
@gabre we would want to check userGivenID's uniqueness right about here I think (under a lock to be safe probably)